File: data/readable_format/fortran/file057.f90
module noseHoover
    implicit none
    integer :: nnos
    real :: kT, gkT, gT
    real, allocatable, dimension(:) :: xi
    real, allocatable, dimension(:) :: v_xi
    real, allocatable, dimension(:) :: Q_i
    real, allocatable, dimension(:) :: G_i
    real, allocatable, dimension(:) :: omega
    logical :: debug
end module noseHoover

program testNoseHoover
    use noseHoover
    implicit none
    
    nnos = 5
    kT = 1.0
    gkT = 2.0
    gT = 3.0
    debug = .true.

    allocate(xi(nnos))
    allocate(v_xi(nnos))
    allocate(Q_i(nnos))
    allocate(G_i(nnos))
    allocate(omega(nnos))

    xi = 1.0
    v_xi = 2.0
    Q_i = 3.0
    G_i = 4.0
    omega = 5.0

    print *, 'nnos =', nnos
    print *, 'kT =', kT
    print *, 'gkT =', gkT
    print *, 'gT =', gT
    print *, 'xi(1) =', xi(1)
    print *, 'v_xi(1) =', v_xi(1)
    print *, 'Q_i(1) =', Q_i(1)
    print *, 'G_i(1) =', G_i(1)
    print *, 'omega(1) =', omega(1)
    print *, 'debug =', debug

    deallocate(xi, v_xi, Q_i, G_i, omega)
end program testNoseHoover

File: data/readable_format/cpp/file057.cpp
#include <iostream>
#include <vector>

// Define the noseHoover class directly in this file
class noseHoover {
public:
    int nnos;
    float kT, gkT, gT;
    std::vector<float> xi;
    std::vector<float> v_xi;
    std::vector<float> Q_i;
    std::vector<float> G_i;
    std::vector<float> omega;
    bool debug;

    // Constructors, destructors, and other member functions can be added here
};

int main() {
    noseHoover nh;

    nh.nnos = 5;
    nh.kT = 1.0f;
    nh.gkT = 2.0f;
    nh.gT = 3.0f;
    nh.debug = true;

    nh.xi.resize(nh.nnos, 1.0f);
    nh.v_xi.resize(nh.nnos, 2.0f);
    nh.Q_i.resize(nh.nnos, 3.0f);
    nh.G_i.resize(nh.nnos, 4.0f);
    nh.omega.resize(nh.nnos, 5.0f);

    std::cout << "nnos = " << nh.nnos << std::endl;
    std::cout << "kT = " << nh.kT << std::endl;
    std::cout << "gkT = " << nh.gkT << std::endl;
    std::cout << "gT = " << nh.gT << std::endl;
    std::cout << "xi[0] = " << nh.xi[0] << std::endl;
    std::cout << "v_xi[0] = " << nh.v_xi[0] << std::endl;
    std::cout << "Q_i[0] = " << nh.Q_i[0] << std::endl;
    std::cout << "G_i[0] = " << nh.G_i[0] << std::endl;
    std::cout << "omega[0] = " << nh.omega[0] << std::endl;
    std::cout << "debug = " << nh.debug << std::endl;

    return 0;
}